domain-models-runtime/src/main/java/org/folio/rest/impl/TenantAPI.java

```diff
@@ -42,70 +44,50 @@
 
   private static final Logger log = LogManager.getLogger(TenantAPI.class);
 
-  private static Map<String, TenantJob> jobs = new HashMap<>();
   private static Map<String, List<Promise<Void>>> waiters = new HashMap<>();
 
   PostgresClient postgresClient(Context context) {
     return PostgresClient.getInstance(context.owner());
   }
 
-  Future<Boolean> tenantExists(Context context, String tenantId) {
-    return Future.future(promise -> tenantExists(context, tenantId, promise));
-  }
-
-  void tenantExists(Context context, String tenantId, Handler<AsyncResult<Boolean>> handler){
+  Future<Boolean> tenantExists(Context context, String tenantId){
     /* connect as user in postgres-conf.json file (super user) - so that all commands will be available */
-    postgresClient(context).select(
-      "SELECT EXISTS(SELECT 1 FROM pg_namespace WHERE nspname = '"+ PostgresClient.convertToPsqlStandard(tenantId) +"');",
-        reply -> {
+    return postgresClient(context).select(
+        "SELECT EXISTS(SELECT 1 FROM pg_namespace WHERE nspname = '"
+            + PostgresClient.convertToPsqlStandard(tenantId) +"');")
+        .compose(result -> {
           try {
-            if(reply.succeeded()){
-              Boolean aBoolean = reply.result().iterator().next().getBoolean(0);
-              handler.handle(io.vertx.core.Future.succeededFuture(aBoolean));
-            }
-            else {
-              log.error(reply.cause().getMessage(), reply.cause());
-              handler.handle(io.vertx.core.Future.failedFuture(reply.cause().getMessage()));
-            }
+            Boolean aBoolean = result.iterator().next().getBoolean(0);
+            return Future.succeededFuture(aBoolean);
           } catch (Exception e) {
             log.error(e.getMessage(), e);
-            handler.handle(io.vertx.core.Future.failedFuture(e.getMessage()));
+            return Future.failedFuture(e.getMessage());
           }
-    });
+        });
```

Comment:
Replace compose (intended for asynchronous operations) by map (intended for synchronous operations) to extract the boolean from the result.
Processing methods like compose and map have built-in try-catch error handling: https://github.com/folio-org/raml-module-builder/blob/master/doc/futurisation.md#result-and-error-handling
Simply use onFailure for logging.

        .map(result -> result.iterator().next().getBoolean(0))
        .onFailure(e -> log.error(e.getMessage(), e));

Reply:
Yes. Much nicer.